cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -1508,22 +1521,25 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Send the ump signals
-	fn send_ump_signal() {
-		use cumulus_primitives_core::relay_chain::{UMPSignal, UMP_SEPARATOR};
+	fn send_ump_signals() {
+		// Take the pending UMP signals.
+		let mut ump_signals = PendingUpwardSignals::<T>::take();
+		// Append the core selector signal.
+		if let Some(core_info) =
+			CumulusDigestItem::find_core_info(&frame_system::Pallet::<T>::digest())
+		{
+			ump_signals.push(
+				UMPSignal::SelectCore(core_info.selector, core_info.claim_queue_offset).encode(),
```

Comment:
We should double check before merging if `UMPSignal::ApprovedPeer` is supported by current RC runtimes.

Reply:
If I understand correctly the `UMPSignal::ApprovedPeer` signal was added in `stable2506`. The RC runtimes have been updated to >= `unstable2507` for the AHM migration. And the validator nodes as well. 89% of the validators have a version > `stable2506`